packages/integration-test/src/setup.js

```diff
@@ -0,0 +1,174 @@
+} = require('@aztec/dev-utils');
+const TruffleContract = require('@truffle/contract');
+
+const { capitaliseFirstChar } = require('./utils');
+
+class Setup {
+    /**
+     * Sets up the environment for an integration test. Specifically, it creates Truffle Contract instances
+     * from the aztec contract-artifacts; at the appropriate address in the contract-addresses package.
+     *
+     * Can be used to setup a local development testing environment for an integration test
+     *
+     * @param {Object} config - config used to modify the test environment setup
+     * @param {String} accounts - Ethereum addresses to be used in the test
+     */
+    constructor(config, accounts) {
+        this.config = config;
+        this.NETWORK = config.NETWORK;
+        this.runAdjustSupplyTests = config.runAdjustSupplyTests;
+        this.runUpgrade = config.runUpgrade;
+        this.contractsToDeploy = config.contractsToDeploy;
+
+        this.provider = web3.currentProvider;
+        this.accounts = accounts;
+
+        const networks = ['ropsten', 'rinkeby', 'kovan', 'mainnet'];
+        if (networks.includes(this.NETWORK)) {
+            console.log('Configuring integration test for', this.NETWORK, 'network');
+            this.networkId = networkIDs[capitaliseFirstChar(this.NETWORK)];
+            this.getAddresses();
+            this.getContractPromises();
+            this.getTransactionTestingAddresses();
+        } else {
+            throw new AztecError(codes.UNSUPPORTED_NETWORK, {
+                message: 'Network not recognised, please input: `ropsten`, `rinkeby`, `kovan` or `mainnet`',
+                inputNetwork: this.NETWORK,
+            });
+        }
+    }
+
+    /**
+     * @method getAddresses() - get the addresses for the desired contracts to deploy
+     */
+    getAddresses() {
+        try {
+            this.contractAddresses = getContractAddressesForNetwork(this.networkId);
+        } catch (err) {
+            throw new AztecError(codes.NO_CONTRACT_ADDRESSES, {
+                message: 'No contract addresses exist for the given network ID in @aztec/contract-addresses',
+                networkId: this.networkId,
+            });
+        }
+    }
+
+    /**
+     * @method getContractPromises - get the Truffle contracts for desired contracts, with the provider set
+     */
+    getContractPromises() {
+        this.contractPromises = this.contractsToDeploy.map((nameOfContract) => this.getContractPromise(nameOfContract));
+    }
+
+    /**
+     * @method getContractPromise - get an object whose key is the name of a contract, and the value is a function that
+     * returns the Truffle representation of the contract
+     * @param {string} nameOfContract
+     */
+    getContractPromise(nameOfContract) {
+        const contractObject = {};
+        let extractedContractArtifact;
+        let extractedContractAddress;
+
+        try {
+            extractedContractArtifact = contractArtifacts[nameOfContract];
```

Comment:
Does this ever throw? Unless I'm missing something, this should just return `undefined` in the event that `nameOfContract` is not in object?

Reply:
You're right. Have refacted to check if `undefined` and throw if so